src/Renderer/QueryResultsRenderer.ts

```diff
@@ -220,28 +220,20 @@ export class QueryResultsRenderer {
         });
 
         for (const [taskIndex, task] of tasks.entries()) {
-            if (this.alreadyRendered(task, renderedTasks)) {
-                continue;
-            }
-
-            if (this.willBeRenderedLater(task, renderedTasks, tasks)) {
-                continue;
-            }
-
-            const listItem = await this.addTaskOrListItem(
-                taskList,
-                taskLineRenderer,
-                task,
-                taskIndex,
-                queryRendererParameters,
-            );
-            renderedTasks.add(task);
-
-            if (task.children.length > 0) {
-                await this.createTaskList(task.children, listItem, queryRendererParameters, renderedTasks);
-                task.children.forEach((childTask) => {
-                    renderedTasks.add(childTask);
-                });
+            if (this.query.queryLayoutOptions.hideChildren) {
+                if (task instanceof Task) {
+                    await this.addTask(taskList, taskLineRenderer, task, taskIndex, queryRendererParameters);
+                }
+            } else {
+                await this.addTaskOrListItemAndChildren(
```

Comment:
I find that `hideChildren` here is not really "hide children". If true, then
- only tasks are rendered, list items are not renderered (this is the missing part)
- the rendering is done without taking into account the parent-child relationships

So I suggest a more honest option like `tree`: `show tree`, `hide tree`. The word "tree" contains inside relationships + list items.

Otherwise I find the implementation very nice =)

Reply:
Ooohhh - I hadn't thought of `tree` - I really like it. Will convert to that. Thanks.